### DISCUSSION TITLE
How does --folder option work when the same pic is in and out of the specified folder?

### DISCUSSION THREAD

**Discussion by oPromessa** [RhetTbull/osxphotos#610] [Q&A] [answered]
Wondering about the `--folder` option semantic.
Looked around for an answer but did not find it. Sorry if this is answered/explained somewhere.

* To start off I understand on Photos you can only create folders under "My Albums". So when it is said "top level folder" on the `--folder` option I believe they are the folders under "My Albums"
* So let's say I have the following structure and the *same pic* `pic001.jpg` is found on two Albums
   * My Albums [Folder]
      * iPhoto Events [Folder]
         * Event01 [Album]
            * pic001.jpg
      * Slideshows [Folder]
         * Show01 [Album]
            *  pic001.jpg
* By using the `--folder "iPhoto Events"` and `--directory "{album[/,.|:,.]}"`options I'm getting the following exported structure:
```
Event01/pic001.jpg
Show01/pic001.jpg
```
* when I was expecting:
 ```
Event01/pic001.jpg
```

Is this because the same pic is on both albums and, **once the pic is selected** it is exported on **all albums it belongs** whether or not they are part of the specified `--folder`?

**Comment by RhetTbull** [comment, marked as answer]
>  So when it is said "top level folder" on the --folder option I believe they are the folders under "My Albums"

Yes

> Is this because the same pic is on both albums and, once the pic is selected it is exported on all albums it belongs whether or not they are part of the specified --folder

Yes, that's how it currently works.  I realize that's somewhat counterintuitive but that is currently how all the query options work with the template options. The query options (in this case `--folder`) select which photos to actually export and the templates (like `--directory "{album[/,.|:,.]}"`) are applied to each of the selected photos on export.

Changing this behavior would require making the template system aware of every query option and substantially complicate the code.  If this doesn't meet your use case, one option would be to create your own template plugin with the `{function}` template.  You could also use `--query-function` to use your own query plugin to work together with the `{function}` plugin.

**Comment by oPromessa** [comment]
Hi there!

Given the  new **filter** functionality from #759 which is now available on [v0.51.2](https://github.com/RhetTbull/osxphotos/releases/tag/v0.51.2) I'm recovering this thread as I believe it would address this topic.

* And it works **GREAT** with the following entry on the **TOML** file:
```toml
# My albums always start with the YYYY format. So 2 means the millennium
directory = "{album|filter(startswith 2)}"
```

* But as _**I'm also performing search/replace on the album name:**_ `directory = "{album[/,.|:,.]"`

* I've  combine them to first filter and then perform the search/replace
```toml
directory = "{album|filter(startswith 2)[/,.|:,.]}"
```

I will now test it out with the the full Library!

**Comment by oPromessa** [comment]
Just a further update in case you want to use `startswith` to match album starting with either 1 or 2:
```toml
directory = "{album|filter(startswith 1|2)[/,.|:,.]}"
```
